core/app/models/spree/calculator/flat_percent_item_total.rb

```diff
@@ -5,9 +5,11 @@
 module Spree
   class Calculator::FlatPercentItemTotal < Calculator
     preference :flat_percent, :decimal, default: 0
+    preference :currency, :string, default: ->{ Spree::Config[:currency] }
 
     def compute(object)
-      computed_amount = (object.amount * preferred_flat_percent / 100).round(2)
+      currency_exponent = ::Money::Currency.find(preferred_currency).exponent
```

Comment:
Shouldn't be the currency taken from the order where the calculator lives instead?

Reply:
Yeah, I like that better. I'll make the changes.